### ISSUE TITLE
[source-google-analytics-data-api]  not taking multiple property Ids

### ISSUE THREAD

**Issue by ali-al-najjar** [airbytehq/airbyte#42464]
### Connector Name

source-google-analytics-data-api

### Connector Version

2.4.13

### What step the error happened?

During the sync

### Relevant information

I added multiple property ids as an array in the property ids field. The connector extracts just the first one.
So for example these two ids, 1738294 and 5729978930 it will get the data for 1738294.
If I switch between them it will get data for the 5729978930 and not the other. I believe there is an issue with the for loop for the property ids.
Last point that this is not something new, I checked all the versions of the source and tested them and nothing was working for all the ids.

### Contribute

- [ ] Yes, I want to contribute

**Comment by marcosmarxm** [comment]
@airbytehq/dev-python can someone take a look into this?

**Comment by girarda** [comment]
@darynaishchenko @maxi297 @strosek as incoming OC - can one of you take a look?

**Comment by strosek** [comment]
Taking a look ...

**Comment by alenoir** [comment]
Hi, any updates on this issue? We’re still experiencing the same problem with multiple property_id values. Thanks!

**Comment by alenoir** [comment]
"Do you know if there is a connection with this scenario to review?" what do you mean ?

I'm running open source version of Airbyte with `airbyte/source-google-analytics-data-api@2.5.0` (same with 2.4.14)

**Comment by aldogonzalez8** [comment]
@alenoir also If you run the read command from your terminal, can you manually add the streams with your property name as a suffix to the catalog streams?

![image](https://github.com/user-attachments/assets/1b9bb293-84a7-447b-8c85-a0230b60cf66)

**Comment by alenoir** [comment]
@aldogonzalez8 I did a "review changes" and everything appeared fine thanks!
Does this mean that I have one table per property (daily_active_users_propertyxxxxxx1, daily_active_users_propertyxxxxxx2) and that I have to activate all the streams for all the properties? Or are all the streams merged into the "daily_active_users" stream? I use the BigQuery destnation.
For the Google Ads stream, all accounts are merged into a single table, is the behavior different?

Translated with DeepL.com (free version)

**Comment by ali-al-najjar** [comment]
@aldogonzalez8 I just tested the connector from Airbyte's UI and attached is the screenshot however I agree with @alenoir that I expected to behave the same as google ads , google search console and google page speed connectors where we can add comma separated urls or properties and all data will be merged in one table using one stream.
<img width="1486" alt="Screenshot 2024-08-13 at 12 01 30 PM" src="https://github.com/user-attachments/assets/024ddc27-bcab-4791-a29b-dba069e0518d">

**Comment by aldogonzalez8** [comment]
> @aldogonzalez8 I did a "review changes" and everything appeared fine thanks! Does this mean that I have one table per property (daily_active_users_propertyxxxxxx1, daily_active_users_propertyxxxxxx2) and that I have to activate all the streams for all the properties? Or are all the streams merged into the "daily_active_users" stream? I use the BigQuery destnation. For the Google Ads stream, all accounts are merged into a single table, is the behavior different?


@alenoir  @ali-al-najjar  Yes, there is an open ticket to change this behavior in the future [here](https://github.com/airbytehq/airbyte-internal-issues/issues/2143). Regarding your question, I would expect to have separate tables in the destination, but I can try to make a test about this.

**Comment by aldogonzalez8** [comment]
> > @aldogonzalez8 I did a "review changes" and everything appeared fine thanks! Does this mean that I have one table per property (daily_active_users_propertyxxxxxx1, daily_active_users_propertyxxxxxx2) and that I have to activate all the streams for all the properties? Or are all the streams merged into the "daily_active_users" stream? I use the BigQuery destnation. For the Google Ads stream, all accounts are merged into a single table, is the behavior different?
> 
> @alenoir @ali-al-najjar Yes, there is an open ticket to change this behavior in the future [here](https://github.com/airbytehq/airbyte-internal-issues/issues/2143). Regarding your question, I would expect to have separate tables in the destination, but I can try to make a test about this.

Yes, for now, we should have separate tables for each stream and property.

**Comment by aldogonzalez8** [comment]
Sorry, just noticed I shared an internal issue, this is the work that will be done:

As of now, the Google Analytics Data API Source supports multiple input property IDs.
To make it work, we produce one stream per property. This means we have ~57 default streams + custom streams, but every time we append a property to the input config, we get 57 more streams. This is better than setting up a connection per property, but for the sake of user experience, we should have a constant number of streams independent of the number of properties. The most obvious way to do it is utilize extra slicing and extend stream schemas with additional column with a property ID as a value.
